File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/FileCenter.m
//
//  FileCenter.m
//  OpenVideoCall
//
//  Created by 3 on 2020/12/15.
//  Portions Copyright (c) 2020 meta-rti. All rights reserved.
//

#import "FileCenter.h"

@implementation FileCenter
+ (NSString *)logDirectory {
    NSString *directory = [[self documentDirectory] stringByAppendingPathComponent:@"MetaLogs"];
    [self checkAndCreateDirectoryAtPath:directory];
    return directory;
}

+ (NSString *)logFilePath {
    return [[self logDirectory] stringByAppendingPathComponent:@"Meta-rtc.log"];
}

+ (NSString *)audioFilePath {
    return [[NSBundle mainBundle] pathForResource:@"audioTest" ofType:@"mp3"];
}

+ (NSString *)documentDirectory {
    return NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES).firstObject;
}

+ (void)checkAndCreateDirectoryAtPath:(NSString *)path {
    BOOL isDirectory = NO;
    BOOL exists = [[NSFileManager defaultManager] fileExistsAtPath:path isDirectory:&isDirectory];
    
    if (!exists || !isDirectory) {
        [[NSFileManager defaultManager] createDirectoryAtPath:path withIntermediateDirectories:YES attributes:nil error:nil];
    }
}
@end

File: Group-Video/OpenVideoCall-iOS/OpenVideoCall/FileCenter.swift
//
//  FileCenter.swift
//  OpenVideoCall
//
//  Created by 3 on 2020/12/16.
//  Portions Copyright (c) 2020 meta-rti. All rights reserved.
//

import Foundation

class FileCenter {
    static func logDirectory() -> String {
        let directory = documentDirectory() + "/Portions Copyright (c) 2020 meta-rti. All rights reserved.Logs"
        checkAndCreateDirectory(at: directory)
        return directory
    }
    
    static func logFilePath() -> String {
        return logDirectory() + "/Portions Copyright (c) 2020 meta-rti. All rights reserved.-rtc.log"
    }
    
    static func audioFilePath() -> String {
        return Bundle.main.path(forResource: "audioTest", ofType: "mp3")!
    }
}

private extension FileCenter {
    static func documentDirectory() -> String {
        return NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first!
    }
    
    static func checkAndCreateDirectory(at path: String) {
        var isDirectory: ObjCBool = false
        let exists = FileManager.default.fileExists(atPath: path, isDirectory: &isDirectory)
        if !exists || !isDirectory.boolValue {
            try? FileManager.default.createDirectory(atPath: path, withIntermediateDirectories: true, attributes: nil)
        }
    }
}
